### PR TITLE
Fixes #2135

### LINKED ISSUE #2135: TableFormulaWithXArgument => If it points to a constant parameter, we always return the value, even if the xArgs cannot be resolved?
follow up of https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/2793

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
@Yuri05 Quite a few places where an error is thrown lol

File: src/OSPSuite.Core/Domain/Formulas/Formula.cs [hunk 1]
```diff
@@ -95,14 +95,23 @@ public virtual void ResolveObjectPathsFor(IEntity dependentEntity)
...
+            {
+               objectReference.Object.PropertyChanged += onReferencePropertyChanged;
+               _objectReferences.Add(objectReference);
+            }
          }
       }
 
+      protected virtual void ValidateObjectReference(ObjectReference objectReference, FormulaUsablePath reference, IEntity dependentEntity)
+      {
+         if (objectReference.Object == null)
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Formulas/Formula.cs:111]: default behavior as before.

File: src/OSPSuite.Core/Domain/Formulas/TableFormulaWithXArgument.cs [hunk 2]
```diff
@@ -67,6 +63,15 @@ protected override double CalculateFor(IReadOnlyList<IObjectReference> usedObjec
          return tableFormula.ValueAt(xArgumentObject.Value);
       }
 
+      protected override void ValidateObjectReference(ObjectReference objectReference, FormulaUsablePath reference, IEntity dependentEntity)
+      {
+         //We do not want to throw a validation error if the alias is the XArgs as it can be undefined 
+         if (reference.Alias == XArgumentAlias)
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Formulas/TableFormulaWithXArgument.cs:69]: special behavior for XArgument alias we won't check

File: src/OSPSuite.Core/Domain/Services/ModelValidator.cs [hunk 3]
```diff
@@ -66,22 +67,26 @@ protected void CheckFormulaIn(IUsingFormula entity, IFormula formulaToCheck, Res
...
+               addNotificationType(NotificationType.Error, objectWithError, Validation.FormulaIsBlackBoxIn(entity.Name, entityAbsolutePath));
+               return;
 
-         foreach (var objectPath in formulaToCheck.ObjectPaths)
-         {
-            CheckPath(entity, objectPath, resolveErrorBehavior);
+            //Table formula with offset may use an x argument that is not defined. A dynamic check will be done in the formula itself
+            //so in this case, we only check the path to the table that should be defined
+            case TableFormulaWithXArgument formulaWithXArgument:
+               checkPathInEntity(formulaWithXArgument.FormulaUsablePathBy(formulaWithXArgument.TableObjectAlias));
```
Comment (msevestre) [inline src/OSPSuite.Core/Domain/Services/ModelValidator.cs:84]: Also here, during the validation, we will not check all object path. Rather only the Table will be checked

File: tests/OSPSuite.Core.Tests/Domain/TableFormulaWithXArgumentSpecs.cs [hunk 4]
```diff
@@ -97,6 +96,22 @@ public void should_return_the_value_of_the_formula_instead()
...
+      protected override void Context()
+      {
+         base.Context();
+         _xArgumentObject.Name = "Another parameter not found";
+      }
+
+      [Observation]
+      public void should_not_crash()
+      {
+         sut.ResolveObjectPathsFor(_parameter);
```
Comment (msevestre) [inline tests/OSPSuite.Core.Tests/Domain/TableFormulaWithXArgumentSpecs.cs:110]: this crashes without my change

**Comment by msevestre** [comment]
@Yuri05 Quite a few places where an error is thrown lol
I am going to merge but please review

**Review by Yuri05 — COMMENTED** [review]
looks reasonable

**Pull request merged**